flink-core/src/test/java/org/apache/flink/testutils/TestingUtils.java

```diff
@@ -56,4 +56,9 @@ public static ScheduledExecutor defaultScheduledExecutor() {
     public static UUID zeroUUID() {
         return ZERO_UUID;
     }
+
+    public static Duration infiniteDuration() {
+        // a year should be more than enough for simulating an infinite duration
+        return Duration.ofDays(365L);
```

Comment:
nit: I'd still prefer `Long.MAX_VALUE` here. That would make it obvious through the code and we could remove the comment. Additionally, we could move the method up next to `infiniteTime()` since they are closely related and refactor `infiniteTime()` in the following way:
```
public static Time infiniteTime() {
    return Time.fromDuration(infiniteDuration());
}
```

Reply:
`Long.MAX_VALUE` unfortunately does not work because internally it will be converted to nanoseconds. Moreover, when doing calculations like calculating a `Deadline`, then you also do calculations with `now`. Therefore, I've chose the 365 days.

I can move the code closer to `infiniteTime`. Changing `infiniteTime` will probably not change a lot. Therefore, I'd like to keep it as is.

Reply:
I see... could we add the reasoning to the comment then to make that clearer?

Reply:
Sure. I will add it.